packages/components/src/Avatar/src/useImageFallback.ts

```diff
@@ -0,0 +1,87 @@
+import { useIsomorphicLayoutEffect } from "@hopper-ui/styled-system";
+import { useState, type ReactEventHandler, useCallback, useRef, type SyntheticEvent, useEffect } from "react";
+
+export interface ImageFallbackProps {
+    /**
+     * The URL of the first choice image to load.
+     */
+    src?: string;
+    /**
+     * The URL of the fallback image to load if the first choice fails.
+     */
+    fallbackSrc?: string | null;
+    /**
+     * The function to call when the image is loaded.
+     */
+    onLoad?: ReactEventHandler<HTMLImageElement>;
+    /**
+     * The function to call when the image fails to load.
+     */
+    onError?: ReactEventHandler<HTMLImageElement>;
+}
+
+type Status = "loading" | "failed" | "pending" | "loaded";
+type ImageEvent = SyntheticEvent<HTMLImageElement, Event>;
+
+export function useImageFallback(props: ImageFallbackProps): [string, Status] {
+    const { src, fallbackSrc, onError, onLoad } = props;
+    const [status, setStatus] = useState<Status>("pending");
+    const [imageUrl, setImageUrl] = useState<string>(src ?? "");
+
+    useEffect(() => {
+        setStatus(src ? "loading" : "pending");
+    }, [src]);
+
+    const imageRef = useRef<HTMLImageElement | null>();
+
+    const load = useCallback(() => {
+        if (!src) {return;}
+    
+        flush();
+
+        const image = new Image();
+
+        image.src = src; // Start loading the initial image
+        setImageUrl(src);
+
+        image.onload = event => {
+            flush();
+            setStatus("loaded");
+            onLoad?.(event as unknown as ImageEvent);
```

Comment:
comment pending - i'm investigating

`event as unknown as ImageEvent ` sure does fix the typescript error, but do those types really match?

Reply:
it's actually what I saw in nextui

Reply:
i guess nextui does the same thing: 
```
    img.onerror = (error) => {
      flush();
      setStatus("failed");
      onError?.(error as any);
    };
```